src/codeExamples/closeHandlingDrawer.stories.mdx

```diff
@@ -0,0 +1,106 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="code examples/close handling for Drawer" />
+
+# Closing Drawer behaviour
+
+Even though the close handling for a [Drawer](?path=/docs/components-drawer--playground) might be implemented as a simple state change,
+sometimes it might be necessary to prevent the [Drawer](?path=/docs/components-drawer--playground) be closed until some action is finished.
+(e.g. uploading a file or saving a form).
+
+Here is an example of how we usually handle the closing functionality:
+
+```jsx
+export const Component = (args) => {
+  const [isShown, setIsShown] = useState(false);
+
+  return (
+        {isShown && (
+          <Drawer
+            footer={<Footer />}
+            title="Title"
+            onClose={() => setIsShown(false)}
+            {...args}
+          >
+            <Content />
+          </Drawer>
+        )}
+  );
+};
+```
+
+And here is an example of how to prevent a [Drawer](?path=/docs/components-drawer--playground) to be closed when there is an action running and you'll need to wait until it ends:
+
+```jsx
+export const Component = () => {
+  // state used for handling the show/hide Drawer
+  const [isShown, setIsShown] = useState(false);
+
+  // state used for enforce user to not close before the action completes
+  const [showForceCancel, setShowForceCancel] = useState(false);
```

Comment:
`const [isProcessing, setIsProcessing] = useState(false);
  const [displayWarning, setDisplayWarning] = useState(false);

  // onClick handler when user attempts to close the Drawer
  let click = 0;
  const handleClose = () => {
    // on first 
    if (click === 0 && isProcessing) {
        setDisplayWarning(true);
        click ++;
        return;
    } else {
        setIsShown(false);
    }
  };

  // in case there is a server error when we process, the application would become unresponsive
  // in that case isProcessing should be timeouted or show an error and setIsProcessing(false) to allow closing the drawer

  return (
      {isShown && (
        <Drawer
          footer={
              <Stack justify="flex-end">
                <Inline gap="md">
                  // show warning if process is in progress 
                  {setDisplayWarning && <Paragraph>Saving changes is in progress</Paragraph>}
                  <Button onClick={handleClose}>{Cancel</Button>
                  <Button
                    isLoading={isProcessing}
                    onClick={() => {
                      // implement your action
                      setIsShown(false);
                    }}
                  >
                    Your action
                  </Button>
                </Inline>
              </Stack>
          }
          onClose={handleClose}
        >
          Your Drawer content here
        </Drawer>
      )}
  );`